src/multiplex/mod.rs

```diff
@@ -175,43 +124,39 @@ impl<T, B, E> Frame<T, B, E> {
     /// Return the request ID associated with the frame.
     pub fn request_id(&self) -> Option<RequestId> {
         match *self {
-            Frame::Message(id, _) => Some(id),
-            Frame::MessageWithBody(id, _, _) => Some(id),
-            Frame::Body(id, _) => Some(id),
-            Frame::Error(id, _) => Some(id),
+            Frame::Message { id, .. } => Some(id),
+            Frame::Body { id, .. } => Some(id),
+            Frame::Error { id, .. } => Some(id),
             Frame::Done => None,
         }
     }
 
     /// Unwraps a frame, yielding the content of the `Message`.
     pub fn unwrap_msg(self) -> T {
```

Comment:
They are mostly there for tests right now. I would be ok moving them into the test code.